opm/parser/eclipse/EclipseState/EclipseState.cpp

```diff
@@ -99,6 +97,9 @@ namespace Opm {
 
     
     MessageContainer& EclipseState::getMessageContainer() {
+        m_messageContainer.appendMessages(m_tables.getMessageContainer());
```

Comment:
Yes; should really think twice (at least!) before you let a `get()` method change the state of the object.

My suggestion: 
 
1. The `EclipseState` has *only*  a const variety of the `getMessageContainer()`
2. The assembly of messages from the various sub components is done at the end of the `EclipseState` constructor.

That way the division into sub message containers is (close to) an implementation detail - and downstream can choose to only care about the collection of messages coming from `EclipseState`